_posts/2020-08-10-deepfake/deepfake.Rmd

```diff
@@ -0,0 +1,231 @@
+df = attr(facemask, 'faces')
+rectX = (df$x - df$radius) 
+rectY = (df$y - df$radius)
+x = (df$x + df$radius) 
+y = (df$y + df$radius)
+
+# draw with red dashed line the box
+imh  = image_draw(image_read('frame_1.png'))
+rect(rectX, rectY, x, y, border = "red", 
+     lty = "dashed", lwd = 2)
+dev.off()
+
+```
+
+<center>
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body", fig.cap = "[Deepfake detection challenge](https://www.kaggle.com/c/deepfake-detection-challenge/data)"}
+knitr::include_graphics("files/frame_2.png")
+```
+
+</center>
+
+## Face extraction
+
+If face locations are found, then it is very easy to extract them all. 
+
+```{r eval=F, echo=T}
+edited = image_crop(imh, "49x49+66+34")
+edited = image_crop(imh, paste(x-rectX+1,'x',x-rectX+1,'+',rectX, '+',rectY,sep = ''))
+edited
+```
+
+<center>
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body", fig.cap = "[Deepfake detection challenge](https://www.kaggle.com/c/deepfake-detection-challenge/data)"}
+knitr::include_graphics("files/frame_1_face.png")
+```
+
+</center>
+ 
+ 
+## Bringing a solution
+
+After dataset preparation, it is time to build a deep learning model with Keras. We can quickly place all the images to folders and using image generators feed faces to a pre-trained Keras model. 
+
+```{r eval=F,echo=T}
+train_dir = 'fakes_reals'
+width = 150L
+height = 150L
+epochs = 10
+
+train_datagen = image_data_generator(
+  rescale = 1/255,
+  rotation_range = 40,
+  width_shift_range = 0.2,
+  height_shift_range = 0.2,
+  shear_range = 0.2,
+  zoom_range = 0.2,
+  horizontal_flip = TRUE,
+  fill_mode = "nearest",
+  validation_split=0.2
+)
+
+
+train_generator <- flow_images_from_directory(
+  train_dir,                  
+  train_datagen,             
+  target_size = c(width,height), 
+  batch_size = 10,
+  class_mode = "binary"
+)
+
+# Build the model ---------------------------------------------------------
+
+conv_base <- application_vgg16(
+  weights = "imagenet",
+  include_top = FALSE,
+  input_shape = c(width, height, 3)
+)
+
+model <- keras_model_sequential() %>% 
+  conv_base %>% 
+  layer_flatten() %>% 
+  layer_dense(units = 256, activation = "relu") %>% 
+  layer_dense(units = 1, activation = "sigmoid")
+
+model %>% compile(
+  loss = "binary_crossentropy",
+  optimizer = optimizer_rmsprop(lr = 2e-5),
+  metrics = c("accuracy")
+)
+
+history <- model %>% fit_generator(
+  train_generator,
+  steps_per_epoch = ceiling(train_generator$samples/train_generator$batch_size),
+  epochs = 10
+)
+```
+ 
+<center>
+
+<br><br> <a href="https://colab.research.google.com/drive/1Wf9aTdcC_YtigjQIYcG8zJqq2q_vpZR9?usp=sharing" class="colab-root">Reproduce in a <span class="colab-span">Notebook</span></a>
+
+</center>
+
+## Conclusion
+
+This post shows how the R community can work with video data and bring a solution to the computer vision tasks. However, readers should know that the implementation of the following steps may drastically improve model performance:
```

Comment:
please be a bit more specific here: "bring a solution to the computer vision tasks."
(this is a very specific task, with clearly defined subtasks, which you could enumerate here to stress which specific functionality was demonstrated)